### PR TITLE
cpu-flex: Added some necessary code for fullsystem

### PR DESCRIPTION
**Pull request by animalvgamer**
These are features that are necessary for fullsystem simulation.
However, no one has fully tested flexcpu for fullsystem yet.

cpu-flex: Added missing initialization code for fullsystem

Signed-off-by: Bradley Wang <radwang@ucdavis.edu>

cpu-flex: Simple thread-driven commit-time interrupts.

Change-Id: I540c331974cb42ef8985e0dcbe87e1e29f57a0e5
Signed-off-by: Bradley Wang <radwang@ucdavis.edu>

### CODE REVIEW

File: src/cpu/flexcpu/flexcpu_thread.cc [hunk 1]
```diff
@@ -345,6 +346,17 @@ FlexCPUThread::commitAllAvailableInstructions()
 
         inflightInsts.pop_front();
         buffer_shrunk = true;
+
+        if (!interrupts_checked
```
Comment (powerjg) [inline src/cpu/flexcpu/flexcpu_thread.cc]: Can you add a comment describing when interrupts are handled? Is this as soon as they are posted no matter what?

File: src/cpu/flexcpu/flexcpu_thread.cc [hunk 2]
```diff
@@ -913,6 +948,32 @@ FlexCPUThread::onIssueAccessed(weak_ptr<InflightInst> inst)
     inst_ptr->traceData(nullptr);
 #endif
 
+    if (inst_ptr->staticInst()->isLastMicroop()) {
+        DPRINTF(FlexCPUThreadEvent, "Releasing MacroOp after decoding final "
+                                    "microop\n");
+        curMacroOp = StaticInst::nullStaticInstPtr;
+    }
+
+    const bool interrupt_detected = _cpuPtr->checkInterrupts(this);
```
Comment (powerjg) [inline src/cpu/flexcpu/flexcpu_thread.cc]: Why is this needed at issue time?

Comment (animalvgamer) [reply src/cpu/flexcpu/flexcpu_thread.cc]: This is for optimization in the same way branch prediction ends early if we think we can know earlier that an interrupt occurred. We kill any issue logic when an interrupt occurs so that the CPU spends less time issuing, when an interrupt is going to squash all of it anyway on the next time a commit occurs.
